File: lua/mugwump/position.lua
Position = {}
Position.__index = Position

function Position:new(position)
    position = position or {}
    setmetatable(position, self)
    position.__index = self
    position.x = 0
    position.y = 0
    position.found = false

    return position
end

function Position:random_position(max_x, max_y)
    self.x = math.random(max_x - 1)
    self.y = math.random(max_y - 1)
    self.found = false
end

return {
    position = Position
}

File: ruby/mugwump/position.rb
=begin
Class to model the position of a "Mugwump" on a grid
as well as if it has been found or not.
=end
class Position

    attr_accessor :x
    attr_accessor :y
    attr_accessor :found

    # constructor
    def initialize()
        @x = 0
        @y = 0
        @found = false
    end

=begin
    Place the mugwump at a random location on the grid.
    Parameters:
    * max_x - number of horizontal spaces on the grid goes from 0 to one less than the given value.
    * max_y - number of vertical spaces on the grid goes from 0 to one less than the given value.
=end
    def random_position(max_x, max_y)
        @x = rand(max_x)
        @y = rand(max_y)
        @found = false
    end
end